lib/msal-node/test/client/ManagedIdentitySources/CloudShell.spec.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ManagedIdentityApplication } from "../../../src/client/ManagedIdentityApplication";
+import {
+    DEFAULT_SYSTEM_ASSIGNED_MANAGED_IDENTITY_AUTHENTICATION_RESULT,
+    MANAGED_IDENTITY_RESOURCE,
+} from "../../test_kit/StringConstants";
+
+import {
+    ManagedIdentityTestUtils,
+    userAssignedClientIdConfig,
+    managedIdentityRequestParams,
+    systemAssignedConfig,
+} from "../../test_kit/ManagedIdentityTestUtils";
+import { AuthenticationResult } from "@azure/msal-common";
+import { ManagedIdentityClient } from "../../../src/client/ManagedIdentityClient";
+import { ManagedIdentityEnvironmentVariableNames } from "../../../src/utils/Constants";
+import {
+    ManagedIdentityErrorCodes,
+    createManagedIdentityError,
+} from "../../../src/error/ManagedIdentityError";
+
+describe("Acquires a token successfully via an App Service Managed Identity", () => {
+    let OLD_ENVS: NodeJS.ProcessEnv;
+
+    beforeAll(() => {
+        // make a copy of old environment
+        OLD_ENVS = process.env;
+
+        process.env[ManagedIdentityEnvironmentVariableNames.MSI_ENDPOINT] =
+            "msi_IDENTITY_ENDPOINT";
+    });
+
+    afterAll(() => {
+        // restore old environment
+        process.env = OLD_ENVS;
```

Comment:
Does this work in all cases, for example: 

1. old_env = [VAR1 = "foo", VAR2 = "bar"]
2. then you set VAR1 = "baz" and VAR3 = "xyz"
3. then restore

You should the vars at step1 and no VAR3. At least in .NET the team had to do some manual deletions.

Reply:
Thank you! This is the root cause of a bug I have been experiencing during testing <3